### PR TITLE
Add Aspire Component / DI Extensions

### PR DESCRIPTION
**Pull request by luisquintanilla**
This PR:

- Adds DI extensions to register LlamaParseClient with DI.
- Fixes reading InMemory files in Aspire sample.

### CODE REVIEW

**Comment by luisquintanilla** [comment]
@colombod ready for review when you get a chance. Given that this is standard DI, I don't think it needs Aspire branding. Also, instead of having a separate project for this, maybe it's worth just adding another class to the LlamaParse project with these extensions. Thoughts?

File: src/Aspire.LlamaParse.Client/AspireLlamaParseExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace Microsoft.Extensions.Hosting;
+
+public static class AspireLlamaParseExtensions
+{
+    /// <summary>
+    /// Adds a LlamaParseClient to the service collection
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="apiKey">The LlamaCloud API key</param>
+    public static void AddLlamaParseClient(this IHostApplicationBuilder builder, string apiKey)
```
Comment (luisquintanilla) [inline src/Aspire.LlamaParse.Client/AspireLlamaParseExtensions.cs:14]: @colombod should these methods take an endpoint and Configuration as well?

Looking at this constructor for reference. 

https://github.com/colombod/llamaindex.net/blob/019048db80c154f9dc37a4a50f97d1eed7001b2f/src/LlamaParse/LlamaParseClient.cs#L13

Comment (colombod) [reply src/Aspire.LlamaParse.Client/AspireLlamaParseExtensions.cs:14]: I think so

**Review by colombod — APPROVED** [review]

File: src/Aspire.LlamaParse.Client/AspireLlamaParseExtensions.cs (hunk 1 above)
Comment (luisquintanilla) [reply src/Aspire.LlamaParse.Client/AspireLlamaParseExtensions.cs]: Thanks. Updated.

**Pull request merged**